### PR TITLE
gh-120164: Fix ``test_os.test_win32_mkdir_700``

### PR DESCRIPTION
**Pull request by Eclips4**
* Issue: gh-120164

### CODE REVIEW

**Comment by zooba** [comment]
LGTM. I assume it works on your machine?

**Comment by Eclips4** [comment]
> LGTM. I assume it works on your machine?

Yes, it is.

**Review by eryksun — APPROVED** [review]
LGTM. :rocket:

File: Lib/test/test_os.py [hunk 1]
```diff
@@ -1837,9 +1837,10 @@ def test_win32_mkdir_700(self):
         os.mkdir(path, mode=0o700)
         out = subprocess.check_output(["cacls.exe", path, "/s"], encoding="oem")
         os.rmdir(path)
+        out = out.strip().rsplit(" ")[1]
```
Comment (vstinner) [inline Lib/test/test_os.py]: What if the patch contains a space? Maybe use a negative index to count from the end?

Comment (eryksun) [reply Lib/test/test_os.py]: Good point, but you could also just limit the number of splits.

```suggestion
        out = out.strip().rsplit(" ", 1)[1]
```

Comment (vstinner) [reply Lib/test/test_os.py]: What if a path contains two spaces? :-) I don't think that the number of splits should be limited. Just pick the last item, or -2, I didn't test.

Comment (eryksun) [reply Lib/test/test_os.py]: It's an `rsplit()` to get the SDDL string from the end, which will not contain spaces. "cacls.exe" prints the path, a space, and the SDDL string. For example:

```text
>>> os.mkdir('C:\\Temp\\spam', 0o700)
>>> os.system('cacls C:\\Temp\\spam /s')
C:\Temp\spam "D:P(A;OICI;FA;;;SY)(A;OICI;FA;;;BA)(A;OICI;FA;;;OW)"
```

The test compares the SDDL string "D:P(A;OICI;FA;;;SY)(A;OICI;FA;;;BA)(A;OICI;FA;;;OW)". This describes a [P]rotected [D]ACL that [A]llows [F]ull-[A]ccess to the directory, and files (OI) and directories (CI) contained within it, to the accounts [SY]stem, [B]uiltin-[A]dministrators, and [OW]ner-Rights.

Comment (vstinner) [reply Lib/test/test_os.py]: Oh right, I missed the R in rsplit() :-)

**Review by vstinner — APPROVED** [review]
LGTM

**Comment by vstinner** [comment]
Merged, thanks for the fix @Eclips4.

**Comment by Eclips4** [comment]
Thanks for the reviews!

**Pull request merged**